src/modules/search-time/selector/index.tsx

```diff
@@ -43,12 +43,33 @@ export default function SearchTimeSelector({
           };
 
     setSelectedMode(newState);
-
     onChange(newState);
   };
 
+  const resetToCurrentTime = () => {
+    setSelectedTime(() => format(new Date(), 'HH:mm'));
+  };
+
+  const resetToCurrentDate = () => {
+    setSelectedDate(new Date());
+  };
+
+  const isPastDate = (selectedDate: string) => {
+    const today = new Date().toISOString().split('T')[0];
+
+    // If reselecting from (selectedYear > current year) to current year, the date will autmatically be reset to the current date.
+    if (selectedDate.substring(0, 4) < initialDate.getFullYear().toString()) {
+      resetToCurrentDate();
+    }
+
+    // To ensure that the time is not past whenever reselecting back the current date.
+    if (selectedDate <= today) resetToCurrentTime();
+    return selectedDate < today;
```

Comment:
How does this logic work when the comparison is two strings against each other? E.g. `'2023-12-24' < '2023-12-31'`.

Reply:
<img width="221" alt="Screenshot 2023-12-18 at 08 54 43" src="https://github.com/AtB-AS/planner-web/assets/59939294/7e52a4c4-b295-42b9-97b6-912820cbd46a">

Reply:
Amazing, TIL